pkg/controller/dataimportcron-controller_test.go

```diff
@@ -0,0 +1,315 @@
+
+	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+)
+
+var (
+	cronLog         = logf.Log.WithName("data-import-cron-controller-test")
+	cronName        = "test-cron"
+	imageStreamName = "test-imagestream"
+)
+
+const (
+	testDigest    = "sha256:68b44fc891f3fae6703d4b74bcc9b5f24df8d23f12e642805d1420cbe7a4be70"
+	testDockerRef = "quay.io/kubevirt/blabla@" + testDigest
+)
+
+var _ = Describe("All DataImportCron Tests", func() {
+	var _ = Describe("DataImportCron controller reconcile loop", func() {
+		var (
+			reconciler *DataImportCronReconciler
+			cronKey    = types.NamespacedName{Name: cronName, Namespace: metav1.NamespaceDefault}
+			cronReq    = reconcile.Request{NamespacedName: cronKey}
+			cronJobKey = func(cron *cdiv1.DataImportCron) types.NamespacedName {
+				return types.NamespacedName{Name: GetCronJobName(cron), Namespace: reconciler.cdiNamespace}
+			}
+			dataSourceKey = func(cron *cdiv1.DataImportCron) types.NamespacedName {
+				return types.NamespacedName{Name: cron.Spec.ManagedDataSource, Namespace: metav1.NamespaceDefault}
+			}
+			dvKey = func(dvName string) types.NamespacedName {
+				return types.NamespacedName{Name: dvName, Namespace: metav1.NamespaceDefault}
+			}
+		)
+		AfterEach(func() {
+			if reconciler != nil {
+				close(reconciler.recorder.(*record.FakeRecorder).Events)
+				reconciler = nil
+			}
+		})
+
+		It("Should do nothing and return nil when no DataImportCron exists", func() {
+			reconciler = createDataImportCronReconciler()
+			_, err := reconciler.Reconcile(context.TODO(), cronReq)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		DescribeTable("Should", func(setDeletionTimestamp bool) {
+			cron := newDataImportCron(cronName)
+			reconciler = createDataImportCronReconciler(cron)
+			_, err := reconciler.Reconcile(context.TODO(), cronReq)
+			Expect(err).ToNot(HaveOccurred())
+
+			cronjob := &v1beta1.CronJob{}
+			err = reconciler.client.Get(context.TODO(), cronJobKey(cron), cronjob)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = reconciler.client.Get(context.TODO(), cronKey, cron)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(cron.Finalizers).ToNot(BeNil())
+			Expect(cron.Finalizers[0]).To(Equal(dataImportCronFinalizer))
+
+			if setDeletionTimestamp {
+				now := metav1.Now()
+				cron.DeletionTimestamp = &now
+				err = reconciler.client.Update(context.TODO(), cron)
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = reconciler.Reconcile(context.TODO(), cronReq)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = reconciler.client.Get(context.TODO(), cronJobKey(cron), cronjob)
+				Expect(err).To(HaveOccurred())
+			}
+		},
+			Entry("create CronJob and add finalizer on a valid DataImportCron with source registry URL", false),
+			Entry("delete CronJob when DataImportCron with source registry URL has DeletionTimestamp", true),
+		)
+
+		DescribeTable("Should", func(setSucceeded bool) {
```

Comment:
Again this is just a sequence of events you are testing. There is no sense in running the same test twice, but the second time something extra. Just run the second one in an it. The coverage will be the same

Reply:
fixed